packages/web3torrent/src/utils/magnet.ts

```diff
@@ -1,39 +1,34 @@
 import {defaultTrackers, EmptyTorrent} from '../constants';
 import {RoutePath} from '../routes';
 import {Torrent} from '../types';
+import {useLocation} from 'react-router-dom';
 
-export const parseMagnetURL: (rawMagnetURL: string) => Torrent = (rawMagnetURL = '') => {
-  const emptyTorrentData = {...EmptyTorrent, magnetURI: ''} as Torrent;
-  if (!rawMagnetURL.trim()) {
-    return emptyTorrentData;
-  }
-  const magnetParams = new URLSearchParams(rawMagnetURL.trim().replace(/#magnet:/g, ''));
-  if (!magnetParams.has('xt')) {
-    return emptyTorrentData;
-  }
+export const useQuery = () => {
+  return new URLSearchParams(useLocation().search);
+};
 
-  if (!magnetParams.has('tr')) {
-    defaultTrackers.map(tr => magnetParams.append('tr', tr));
+export const parseURL: (infoHash: string, queryParams: any) => Torrent = (
+  infoHash,
+  queryParams
+) => {
+  if (!infoHash || !infoHash.trim()) {
+    return EmptyTorrent;
   }
+  const name = queryParams.get('name') || queryParams.get('dn') || 'unknown';
+  const length = Number(queryParams.get('length')) || 0;
+  const magnetURI = buildMagnetURI(name, length, infoHash, defaultTrackers);
+  return {...EmptyTorrent, infoHash, name, length, magnetURI};
+};
 
-  return {
-    ...emptyTorrentData,
-    name: magnetParams.get('name') || magnetParams.get('dn') || 'unknown',
-    magnetURI: decodeURIComponent(`magnet:?${magnetParams.toString()}`),
-    infoHash: (magnetParams.get('xt') as string).substring(9),
-    length: Number(magnetParams.get('xl')) || 0,
-    cost: magnetParams.get('cost') || '0'
-  };
+const buildMagnetURI = (name = 'unknown', length = 0, infoHash, trackers) => {
+  const parameters = new URLSearchParams(`?xt=urn:btih:${infoHash}&dn=${name}&xl=${length}`);
+  trackers.map(tr => parameters.append('tr', tr));
+  return decodeURIComponent(`magnet:?${parameters.toString()}`);
```

Comment:
Not sure I fully understand this. Is there a reason to use `URLSearchParams` object and then immediately convert the object to a string? Would using a dictionary work for the `parameters` variable?

Reply:
Changed to a reduce function, that seems to work just as fine, and it's cleaner and shorter.